pkg/collector/reconcile/opentelemetry.go

```diff
@@ -38,6 +39,18 @@ import (
 func Self(ctx context.Context, params manifests.Params) error {
 	changed := params.Instance
 
+	up := &collectorupgrade.VersionUpgrade{
+		Log:      params.Log,
+		Version:  version.Get(),
+		Client:   params.Client,
+		Recorder: params.Recorder,
+	}
+	changed, err := up.ManagedInstance(ctx, changed)
+	if err != nil {
+		// don't fail to allow setting the status
+		params.Log.Error(err, "failed to upgrade the OpenTelemetry CR")
```

Comment:
we could even add this to the recorder event if we wanted to, but this is fine for now. Thank you!